packages/twenty-front/src/modules/favorites/components/FavoriteFoldersMultiSelect.tsx

```diff
@@ -0,0 +1,211 @@
+import { isFavoriteFolderCreatingState } from '@/favorites/states/isFavoriteFolderCreatingState';
+import { ObjectRecord } from '@/object-record/types/ObjectRecord';
+import { DropdownMenu } from '@/ui/layout/dropdown/components/DropdownMenu';
+import { DropdownMenuItemsContainer } from '@/ui/layout/dropdown/components/DropdownMenuItemsContainer';
+import { DropdownMenuSearchInput } from '@/ui/layout/dropdown/components/DropdownMenuSearchInput';
+import { DropdownMenuSeparator } from '@/ui/layout/dropdown/components/DropdownMenuSeparator';
+import { useDropdown } from '@/ui/layout/dropdown/hooks/useDropdown';
+import { useScopedHotkeys } from '@/ui/utilities/hotkey/hooks/useScopedHotkeys';
+import { useAvailableScopeIdOrThrow } from '@/ui/utilities/recoil-scope/scopes-internal/hooks/useAvailableScopeId';
+import { useTheme } from '@emotion/react';
+import styled from '@emotion/styled';
+import { useCallback } from 'react';
+import { useRecoilState, useRecoilValue } from 'recoil';
+import { Key } from 'ts-key-enum';
+import { IconPlus, MenuItem, MenuItemMultiSelect } from 'twenty-ui';
+import { useDebouncedCallback } from 'use-debounce';
+
+const StyledDropdownContainer = styled.div`
+  position: relative;
+  width: 100%;
+`;
+
+const StyledItemsContainer = styled.div`
+  width: 100%;
+  max-height: 160px;
+  overflow-y: auto;
+
+  &::-webkit-scrollbar {
+    display: none;
+  }
+
+  -ms-overflow-style: none;
+  scrollbar-width: none;
+`;
+
+const StyledFooter = styled.div`
+  background: ${({ theme }) => theme.background.primary};
+  border-bottom-left-radius: ${({ theme }) => theme.border.radius.md};
+  border-bottom-right-radius: ${({ theme }) => theme.border.radius.md};
+  border-top: 1px solid ${({ theme }) => theme.border.color.light};
+  bottom: 0;
+  position: sticky;
+`;
+
+type FavoriteFoldersMultiSelectProps = {
+  onSubmit?: () => void;
+  record?: ObjectRecord;
+  objectNameSingular: string;
+};
+
+const StyledIconPlus = styled(IconPlus)`
+  padding-left: ${({ theme }) => theme.spacing(2)};
+  padding-right: ${({ theme }) => theme.spacing(1)};
+`;
+
+const StyledDropdownMenuSeparator = styled(DropdownMenuSeparator)`
+  margin-bottom: ${({ theme }) => theme.spacing(1)};
+  margin-top: ${({ theme }) => theme.spacing(1)};
+`;
+
+const NO_FOLDER_ID = 'no-folder';
+
+export const FavoriteFoldersMultiSelect = ({
+  onSubmit,
+  record,
+  objectNameSingular,
+}: FavoriteFoldersMultiSelectProps) => {
+  const [isFavoriteFolderCreating, setIsFavoriteFolderCreating] =
+    useRecoilState(isFavoriteFolderCreatingState);
+
+  const theme = useTheme();
+
+  const scopeId = useAvailableScopeIdOrThrow(
+    FavoriteFoldersScopeInternalContext,
+  );
+
+  const { closeDropdown } = useDropdown(FAVORITE_FOLDERS_DROPDOWN_ID);
+
+  const {
+    favoriteFoldersSearchFilterState,
+    favoriteFoldersMultiSelectCheckedState,
+  } = useFavoriteFoldersScopedStates();
```

Comment:
we don't use this pattern anymore. I can help refactoring that for the example. We now use useRecoilValueV2 (&co), no need for useXXXScopedStates() anymore

Reply:
see AnalyticsActivityGraph which our last (and most simple example). This will greatly simplify your code